### PR TITLE
use shared node type

### PR DESCRIPTION
**Pull request by kangyili**
### What does this PR do?

Move nodes types from pkg/orchestrator/util.go to https://github.com/DataDog/agent-payload/pull/199 so it can be shared with backend side

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/orchestrator/stats.go [hunk 1]
```diff
@@ -16,17 +17,17 @@ var (
...
-	// CacheHits contains the number of cache hits for a NodeType per run.
+	// CacheHits contains the number of cache hits for a agentModel.K8SResources per run.
 	CacheHits int
 
-	// CacheMiss contains the number of cache miss/send Data for a NodeType per run.
+	// CacheMiss contains the number of cache miss/send Data for a agentModel.K8SResources per run.
 	CacheMiss int
 
-	NodeType
+	agentModel.K8SResources
```
Comment (vickenty) [inline pkg/orchestrator/stats.go:26]: Would it make sense to keep the field name?
```suggestion
	NodeType agentModel.K8SResources
```
K8SResources is plural, which doesn't work very well for a field that contains one value only. (or `K8SResource`)

File: pkg/orchestrator/util.go [hunk 2]
```diff
@@ -6,177 +6,38 @@
...
-		return "CustomResource"
-	case K8sUnsetType:
-		return "UnsetType"
-	default:
-		_ = log.Errorf("Trying to convert unknown NodeType iota: %d", n)
-		return "Unknown"
+func NodeTypes() []agentModel.K8SResources {
+	resources := make([]agentModel.K8SResources, 0, len(agentModel.K8SResources_value))
+	for k, v := range agentModel.K8SResources_value {
+		if k != agentModel.K8SResources_UNSET_NODE_TYPE.String() {
```
Comment (vickenty) [inline pkg/orchestrator/util.go:22]: Would this work?
```suggestion
		if v != agentModel.K8SResources_UNSET_NODE_TYPE {
```

**Review by vickenty — APPROVED** [review]
Left a couple of small suggestions, but otherwise LGTM

**Review by xornivore — APPROVED** [review]
LGTM overall but you'll need to resolve conflicts since the common code has changed in the process agent

File: pkg/orchestrator/stats.go (hunk 1 above)
Comment (kangyili) [reply pkg/orchestrator/stats.go:26]: Thanks for the feedback, I updated it to `agentModel.K8SResource`

File: pkg/orchestrator/util.go (hunk 2 above)
Comment (kangyili) [reply pkg/orchestrator/util.go:22]: Addressed the it, it works 👍

**Pull request closed without merging**